### PR TITLE
[BPK-1106] Add withDivider HOC to RN card

### PR DESCRIPTION
**Pull request by matthewdavidson**
So the plan is to remove `react-native-bpk-component-ticket` in favor of a "configuration" on the card component.

I've opted to split this work into 2 PR's:
1. Add the "configuration" to the card to achieve a divider line 
2. Remove the ticket component (see #290)

This is the first PR. I've opted to achieve the divider line using a HOC.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -15,6 +15,9 @@
 - bpk-component-button:
   - Svg's fill colour is automatically set to text colour.
 
+- react-native-bpk-component-card:
+  - Added `withDivider` HOC, see http://backpack.prod.aws.skyscnr.com/components/native/cards/#with-divider
```
Comment (matthewdavidson) [inline changelog.md]: Note these docs don't exist (yet...)

File: native/packages/react-native-bpk-component-card/readme.md [hunk 2]
```diff
@@ -62,11 +62,90 @@ export default class App extends Component {
 }
 ```
 
+### `withDivider` HOC
+
+```js
+import React, { Component } from 'react';
+import { View, StyleSheet, Text } from 'react-native';
+import BpkText from 'react-native-bpk-component-text';
+import * as TOKENS from 'bpk-tokens/tokens/ios/base.react.native.es6';
```
Comment (k0nserv) [inline native/packages/react-native-bpk-component-card/readme.md]: We don't wanna encourage this type of import right?

File: native/packages/react-native-bpk-component-card/src/withDivider.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  View,
+  StyleSheet,
+  ViewPropTypes,
+} from 'react-native';
```
Comment (k0nserv) [inline native/packages/react-native-bpk-component-card/src/withDivider.js]: Could we order these alphabetically?

File: native/packages/react-native-bpk-component-card/readme.md (hunk 2 above)
Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-card/readme.md]: Indeed - fixed.

File: native/packages/react-native-bpk-component-card/src/withDivider.js (hunk 3 above)
Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-card/src/withDivider.js]: We could. Prettier integration seems like a good xmas break task.

Comment (k0nserv) [reply native/packages/react-native-bpk-component-card/src/withDivider.js]: Sounds like a plan yeah

**Pull request merged**